scipy/optimize/minpack.h

```diff
@@ -55,16 +54,16 @@ the result tuple when the full_output argument is non-zero.
   /* Set up callback functions */ \
   if (!PyCallable_Check(fun)) \
     PYERR(errobj,"First argument must be a callable function."); \
-  multipack_python_function = fun; \
-  multipack_extra_arguments = arg; }
+  if (init_callback(&callback, fun, arg) != 0) \
+    PYERR(errobj,"Could not init callback");\
+  callback_inited = 1; \
+  if (setjmp(callback.error_buf) != 0) \
+    PYERR(errobj,"Could not setjmp callback error_buf");\
```

Comment:
`setjmp` never fails, but it can return nonzero due to a longjmp.

Since we are using MINPACK's `iflag` mechanism to bail out when e.g. python calls fail, there's no need to use `setjmp` here, so this should be removed.

Ditto for below.